search/LocalSearch_First_NQueensProblem.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ *@file LocalSearch_First_NQueensProblem.cpp
+ *@autor dhernandezl
+ *@brief [Heuristic Algorithm local Search]
+ * (T. Stützle and R. Ruiz, “Iterated local search,” Handb. Heuristics, vol. 1–2, pp. 579–605, 2018, isbn.9783319071244, doi.10.1007/978-3-319-07124-4_8.)
+ * (https://en.wikipedia.org/wiki/Local_search_(optimization))
+ */
+
+/*
+* The LS algorithm is the basis of many methods used for optimization problems. 
+* Your process can be seen as iterative starting with an initial solution and 
+* improving it by making local modifications.  Its system is based on the search 
+* for neighborhoods that obtain a better solution.
+* The neighborhood search and pivot rule used for replacement are usually:
+*  - Select the best neighbor of all (best-improvement rule).
+*  - Select the first neighbor that improves the solution (first-improvement rule).
+* The rule applied in this file is first.improvement.
+*/
+
+/*
+* Problem solving the algorithm
+* The eight queens problem is a very popular strategy that consists of placing eight 
+* queens within a chessboard without threatening them. Finding the solution for different 
+* dimensions is the same as finding the solution for a minimal problem.
+*/
+
+#include "LocalSearchFirst.h"
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <list>
+#include <time.h>
+#define NREINAS 8
+#define t_neighbor 15
+using namespace std;
+
+vector<vector<int>> neighbour_sys;
+
+void print_vector(vector<int> s) {
+	for (int x : s) { printf("%d ", x); }
+	printf("\n");
+}
+
+vector<int> initial_solution() {
+	vector<int> solution;
+	srand(time(NULL));

```

Comment:
```suggestion
	srand(time(nullptr));
```
Same in other parts of the code.